core/src/components/datetime/datetime.ios.scss

```diff
@@ -72,6 +72,116 @@
   text-transform: uppercase;
 }
 
+@supports (border-radius: mod(1px, 1px)) {
+  .calendar-days-of-week .day-of-week {
+    /**
+     * When the computed font size is 24px
+     * we want the days of the week to only
+     * show the first letter of each day.
+     *
+     * We use the modulus operator to ensure
+     * that any font size less than 24px is
+     * unchanged (i.e. 16 % 24 = 16). If the
+     * font size is 24px then the remainder
+     * will be 0.
+     */
+    $max-font-size: min(1rem, $datetime-dynamic-font-breakpoint);
+    $remainder: mod($max-font-size, $datetime-dynamic-font-breakpoint);
+
+    /**
+     * From here, we scale the remainder to get
+     * a width value larger than 20px.
```

Comment:
Why do we need to do this? I'm confused about why we can't skip the scaling step and clamp the remainder directly; that would ensure that font sizes < 20px would get a minimum width, and very large font sizes wouldn't bleed out of the container since the width would get clamped at 100%.

I'm especially wondering about font sizes that would lead to a clamped width in-between 20px and 100%. For example, a global font size/remainder of 8px would lead to a `$width` of 80px, which could reasonably fit on wide/landscape displays but seems way too wide for such a small font.

Reply:
The `20px` exists so that when the font size is >=24px the days of the week still appear. Since the `$remainder` at that point will be 0, doing 0 * 10 still gives you 0, so 20px exists to ensure that the first letter of each day of week label shows up.

We need to scale the values for font sizes <24px because otherwise the label width won't be large enough. For example if the font size is 23px, then the `$remainder` will be 23px. However, a font size of 23px looks something like this:

![image](https://github.com/ionic-team/ionic-framework/assets/2721089/67759e60-1ea9-42d2-9d4f-8065969dfac2)

Reply:
Okay, after sleeping on this and playing around with it some more, I think I've got a handle on it now. I was missing a few things:
- The days of the week render at 0.75rem, not 1rem, so a global font size of 24px actually renders these labels at 18px, which is small enough to fit in a 20px width
- The 100% max width kicks in a lot sooner than I was imagining due to the days of the week being in a grid layout
- The days of the week have a max font size, so it's okay that all global font sizes > 24px always result in a width of 20px since the max fits within that, without cutting off parts of the first character

Up to you how much of that you want to include in the comments 😄 Most of it can be sussed out through context, I just had to play around with it to put the pieces together.

Reply:
Thanks! I'll work on getting this reflected in the comments